crates/router/src/core/payment_methods/cards.rs

```diff
@@ -484,6 +512,60 @@ fn filter_installment_based(
     })
 }
 
+async fn filter_payment_country_based(
+    db: &dyn db::StorageInterface,
+    payment_intent: &storage::PaymentIntent,
+    pm: &api::ListPaymentMethodResponse,
+) -> errors::CustomResult<bool, errors::ApiErrorResponse> {
+    let address = helpers::get_address_by_id(db, payment_intent.billing_address_id.clone()).await?;
```

Comment:
This query already called in the same session before?

Reply:
Fixed